grub-reboot-picker/grub-reboot-picker.py

```diff
@@ -32,12 +36,13 @@ def get_all_grub_entries(file_path, include_submenus=True):
         'UEFI Firmware Settings': []
     }
     """
-    with open(file_path, 'r') as file:
-        lines = file.readlines()
 
-    menu_pattern = re.compile("^\\s*menuentry ['\"]([^'\"]*)['\"]")
-    submenu_pattern = re.compile("^\\s*submenu ['\"]([^']*)['\"]")
-    closing_brace_pattern = re.compile("^\\s*}")
+    grub_mkconfig_output = subprocess.check_output(["pkexec", "grub-mkconfig"], stderr=subprocess.STDOUT)
```

Comment:
This is simpler than parsing the grub.cfg file, because grub.cfg is not world readable, which means this Python script had to run as sudo. grub-mkconfig needs sudo too, but via polkit I was able to let it run without a user prompt.